### PR TITLE
add support for calling ECDSA signatures RSA signatures, cuz node allows it

### PR DESCRIPTION
**Pull request by calvinmetcalf**
fix for #32

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
I'm going to go ahead and merge/publish this latter if nobody objects

File: test/index.js [hunk 1]
```diff
@@ -63,6 +63,7 @@ fixtures.valid.ec.forEach(function (f) {
   }
 
   test(f.message, function (t) {
+    console.log(f.scheme)
```
Comment (dcousens) [inline test/index.js]: random `console.log`

File: test/index.js [hunk 2]
```diff
@@ -79,6 +80,27 @@ fixtures.valid.ec.forEach(function (f) {
 
     t.end()
   })
+  if (f.scheme !== 'DSA' && f.scheme.toLowerCase().indexOf('dsa') === -1) {
+    test(f.message + ' named rsa through', function (t) {
+      var scheme = 'RSA-' + f.scheme.toUpperCase()
+      console.log(scheme)
```
Comment (dcousens) [inline test/index.js]: here too

File: browser/sign.js [hunk 3]
```diff
@@ -6,18 +6,18 @@ var BN = require('bn.js')
...
     return ecSign(hash, priv)
   } else if (priv.type === 'dsa') {
     if (signType !== 'dsa') throw new Error('wrong private key type')
     return dsaSign(hash, priv, hashType)
   } else {
-    if (signType !== 'rsa') throw new Error('wrong private key type')
+    if (signType !== 'rsa' && signType !== 'ecdsa/rsa') throw new Error('wrong private key type')
   }
-
+  hash = Buffer.concat([tag, hash])
```
Comment (dcousens) [inline browser/sign.js]: why did you move this in?

File: browser/sign.js [hunk 4]
```diff
@@ -6,18 +6,18 @@ var BN = require('bn.js')
 var parseKeys = require('parse-asn1')
 var curves = require('./curves.json')
 
-function sign (hash, key, hashType, signType) {
+function sign (hash, key, hashType, signType, tag) {
   var priv = parseKeys(key)
   if (priv.curve) {
-    if (signType !== 'ecdsa') throw new Error('wrong private key type')
```
Comment (dcousens) [inline browser/sign.js]: Could you add a comment here about why this is the case?
`ecdsa/rsa` seems strange

**Comment by dcousens** [comment]
LGTM otherwise

**Comment by calvinmetcalf** [comment]
Ok will update

File: browser/sign.js (hunk 3 above)
Comment (calvinmetcalf) [inline browser/sign.js]: ah it's because we only need the tag when it is an rsa key NOT when it's an ecdsa key so since we can't tell based on the signature type asked for we have to do it based on what rutine they are using, so this code path only is executed for rsa signatures never (ec)dsa
